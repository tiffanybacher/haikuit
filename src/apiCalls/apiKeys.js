export const wordsApiKey = 'eb0eafc31cmshf64d24e82c012a0p1c8a93jsnb7c192842282';