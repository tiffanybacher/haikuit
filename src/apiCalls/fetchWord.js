import * as actions from '../actions';

export const fetchWord = (searchQuery) => {
  const wordsApiKey = process.env.REACT_APP_WORDS_KEY;
  
  return async (dispatch) => {
    const url = `https://wordsapiv1.p.mashape.com/words/${searchQuery}`
    const options = {
      headers: { 'X-RapidAPI-Key': `${wordsApiKey}` }
    }

    try {
      dispatch(actions.setLoading(true));

      const response = await fetch(url, options);

      if(!response.ok) {
        throw Error(`"${searchQuery}" not found`);
      }
      
      const data = await response.json();

      dispatch(actions.setLoading(false));

      return data;
    } catch (error) {
      dispatch(actions.setError(error.message));
    }
  }
}