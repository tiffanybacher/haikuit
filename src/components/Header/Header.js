import React, { Component } from 'react';
import HamburgerIcon from '../HamburgerIcon/HamburgerIcon';
import SearchIcon from '../SearchIcon/SearchIcon';

class Header extends Component {
  state = {
    searchIsShown: false
  }

  toggleSearch = () => {
    this.setState({
      searchIsShown: !this.state.searchIsShown
    });
  }

  render() {
    return (
      <header className="Header">
        <div className="flex-container">
          <HamburgerIcon />
          <h1>Haikuit<i class="fas fa-feather-alt"></i></h1>
          <SearchIcon toggleSearch={this.toggleSearch} />
        </div>
        {
        this.state.searchIsShown 
          ? <input 
              type="text" 
              className="search-input" 
              placeholder="Search for a word..." />
          : null
        }
      </header>
    );
  }
}

export default Header;