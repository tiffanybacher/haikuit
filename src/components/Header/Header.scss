.Header {
  background: $purple;
  color: white;
  padding: 10px 15px 15px 15px;
  text-align: center;

  h1 {
    font-family: 'Baloo', sans-serif;
    font-size: 2.5em;
    letter-spacing: 2px;
    margin: 0 15px 15px 15px;
    position: relative;
    top: 10px;
  }

  .search-input {
    @include border;
    font-style: italic;
    padding: 10px;
    width: 50%;
  }

  .flex-container {
    justify-content: space-between;
  }

  .fa-search,
  .fa-bars {
    color: white;
    font-size: 1.75em;
  }
}