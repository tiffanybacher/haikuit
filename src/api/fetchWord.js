import { wordsApiKey } from './apiKeys'

export const fetchWord = async (searchQuery) => {
  const url = `https://wordsapiv1.p.mashape.com/words/${searchQuery}`
  const options = {
    headers: {"X-RapidAPI-Key": `${wordsApiKey}`}
  }

  try {
    const response = await fetch(url, options);

    if(!response.ok) {
      throw Error(response.statusText);
    }
    
    const data = await response.json();

    return data;
  } catch (error) {
    console.log(error.message);
  }
}