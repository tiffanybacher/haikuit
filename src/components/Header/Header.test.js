import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapDispatchToProps } from './Header';
import { fetchWord } from '../../apiCalls/fetchWord';

jest.mock('../../apiCalls/fetchWord');

describe('Header', () => {
  let toggleMenu;
  let menuShown;
  let wrapper;

  beforeEach(() => {
    toggleMenu = jest.fn();
    menuShown = jest.fn();
    wrapper = shallow(
      <Header 
        toggleMenu={toggleMenu} 
        menuShown={menuShown}
        fetchWord={fetchWord} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle search bar', () => {
    expect(wrapper.state('searchShown')).toBe(false);

    wrapper.instance().toggleSearch();

    expect(wrapper.state('searchShown')).toBe(true);
  });

  it('should match snapshot if searchShown is true', () => {
    wrapper.setState({ searchShown: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should open menu drawer when hamburger btn is clicked', () => {
    wrapper.find('.hamburger-icon').simulate('click');

    expect(toggleMenu).toHaveBeenCalled();
  });

  it('should show search bar when search btn is clicked', () => {
    expect(wrapper.state('searchShown')).toEqual(false);

    wrapper.find('.search-icon').simulate('click');

    expect(wrapper.state('searchShown')).toEqual(true);
  });

  it('should update searchQuery on change in search input', () => {
    const mockEvent = { 
      preventDefault: jest.fn(), 
      target: { value: 'particular' } 
    }

    wrapper.setState({ searchShown: true });

    expect(wrapper.state('searchQuery')).toEqual('');

    wrapper.find('.search-input').simulate('change', mockEvent);

    expect(wrapper.state('searchQuery')).toEqual('particular');
  });

  it('should clear the search input', () => {
    wrapper.setState({ searchShown: true });

    wrapper.find('.clear-btn').simulate('click');

    expect(wrapper.state('searchQuery')).toEqual('');
  });

  it('should match snapshot if result is shown', () => {
    wrapper.setState({
      searchShown: true,
      resultShown: true,
      currentResult: { word: 'tenacity' }
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should map fetchWord from dispatch to props', () => {
    const mockDispatch = jest.fn();
    const mockWord = 'tenacity';
    const thunk = fetchWord(mockWord);
    const props = mapDispatchToProps(mockDispatch);

    props.fetchWord(mockWord);

    expect(mockDispatch).toHaveBeenCalledWith(thunk);
  });
});