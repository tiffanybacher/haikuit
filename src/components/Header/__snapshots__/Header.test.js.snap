// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  className="Header"
>
  <div
    className="flex-container"
  >
    <button
      className="hamburger-icon"
      onClick={[MockFunction]}
    >
      <i
        className="fas fa-bars"
      />
    </button>
    <Link
      tabIndex="-1"
      to="/"
    >
      <h1>
        Haikuit
        <i
          className="fas fa-feather-alt"
        />
      </h1>
    </Link>
    <button
      className="search-icon"
      onClick={[Function]}
    >
      <i
        className="fas fa-search"
      />
    </button>
  </div>
</header>
`;

exports[`Header should match snapshot if searchShown is true 1`] = `
<header
  className="Header"
>
  <div
    className="flex-container"
  >
    <button
      className="hamburger-icon"
      onClick={[MockFunction]}
    >
      <i
        className="fas fa-bars"
      />
    </button>
    <Link
      tabIndex="-1"
      to="/"
    >
      <h1>
        Haikuit
        <i
          className="fas fa-feather-alt"
        />
      </h1>
    </Link>
    <button
      className="search-icon"
      onClick={[Function]}
    >
      <i
        className="fas fa-search"
      />
    </button>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="search-input"
      onChange={[Function]}
      placeholder="Search for a word..."
      type="text"
    />
    <button
      className="search-btn"
    >
      GO
    </button>
  </form>
</header>
`;
